server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1147,20 +1211,84 @@ private void doDeleteShardSnapshots(
                 foundIndices,
                 rootBlobs,
                 updatedRepoData,
+                repositoryData,
                 remoteStoreLockManagerFactory,
+                remoteSegmentStoreDirectoryFactory,
                 afterCleanupsListener,
                 idToShardInfoMap
             );
-            asyncCleanupUnlinkedShardLevelBlobs(
-                repositoryData,
-                snapshotIds,
-                writeShardMetaDataAndComputeDeletesStep.result(),
-                remoteStoreLockManagerFactory,
-                afterCleanupsListener
-            );
+            if (isShallowSnapshotV2 == false) {
```

Comment:
How are we covering cases where we delete only 1 of many snapshots of a deleted index ? In that case, we need to run shard level clean up as on all the shards.

Reply:
Can we add IT for this case as well ? 

Lets create multiple snapshots while indexing data . We can check the remote store directory to see if it has decreased post deletion of some of the snapshots , while still not fully deleted .

Reply:
> How are we covering cases where we delete only 1 of many snapshots of a deleted index ? In that case, we need to run shard level clean up as on all the shards.

Added a check on snapshotID. If the snapshot is found in the deletedSnapshots list, we trigger cleanup